### PR TITLE
inject db gradle dependencies using api

### PR DESCRIPTION
**Pull request by mshima**
---

Please make sure the below checklist is followed for Pull Requests.

- [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [ ] Tests are added where necessary
- [ ] The JDL part is updated if necessary
- [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [ ] Documentation is added/updated where necessary
- [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (below reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

File: generators/server/templates/gradle/profile_dev.gradle.ejs [hunk 1]
```diff
@@ -49,7 +41,7 @@ configurations {
 }
 
 dependencies {
-    developmentOnly "org.springframework.boot:spring-boot-devtools:${springBootVersion}"
+    compileOnly "org.springframework.boot:spring-boot-devtools"
```
Comment (murdos) [inline generators/server/templates/gradle/profile_dev.gradle.ejs:44]: Out of curiosity, what's the reason for this change?

Comment (mshima) [reply generators/server/templates/gradle/profile_dev.gradle.ejs:44]: Test with developmentOnly again since its recommended by updated spring-boot.
```suggestion
    developmentOnly "org.springframework.boot:spring-boot-devtools"
```

Comment (mshima) [reply generators/server/templates/gradle/profile_dev.gradle.ejs]: It fails to generate a package with dev profile.
See https://github.com/jhipster/generator-jhipster/actions/runs/8316726401/job/22756588401?pr=25504

**Review by DanielFran — APPROVED** [review]
